src/main/kotlin/de/klg71/keycloakmigration/changeControl/actions/role/AddRoleAction.kt

```diff
@@ -38,8 +44,25 @@ class AddRoleAction(
         }
         // If the action fails from here we have to role it back
         setExecuted()
-        findRole().run {
-            client.updateRole(updateRole(this), id, realm())
+        val role = findRole()
+        client.updateRole(updateRole(role), role.id, realm())
+        if(composite != null && composite && !compositeChildRoles.isNullOrEmpty()) {
+            val roleItems = compositeChildRoles.map {
+                val foundRole = if (it.clientId == null) {
```

Comment:
i would like to refactor the `compositeChilldRoles.map{}`-block into a separate method

Reply:
done. check whether this is better